esphome/components/ili9xxx/ili9xxx_display.cpp

```diff
@@ -302,58 +289,22 @@ void ILI9XXXDisplay::draw_pixels_at(int x_start, int y_start, int w, int h, cons
   // x_ and y_offset are offsets into the source buffer, unrelated to our own offsets into the display.
   if (x_offset == 0 && x_pad == 0 && y_offset == 0) {
     // we could deal here with a non-zero y_offset, but if x_offset is zero, y_offset probably will be so don't bother
-    this->write_array(ptr, w * h * 2);
+    this->bus_->write_array(ptr, w * h * 2);
   } else {
     auto stride = x_offset + w + x_pad;
     for (size_t y = 0; y != h; y++) {
-      this->write_array(ptr + (y + y_offset) * stride + x_offset, w * 2);
+      this->bus_->write_array(ptr + (y + y_offset) * stride + x_offset, w * 2);
     }
   }
-  this->end_data_();
+  this->bus_->end_transaction();
```

Comment:
```suggestion
  this->bus_->end_pixels();
```
I'm missing the start_transaction has well above but that should be this->bus->start_pixels(); ;)